.github/workflows/ci.yml

```diff
@@ -45,23 +54,48 @@ jobs:
         run: |
           sudo apt update
           sudo apt install libpulse-dev libdbus-1-dev libncursesw5-dev libxcb-shape0-dev libxcb-xfixes0-dev
-      - uses: actions/checkout@v2
-        name: Checkout src
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      - name: Running cargo check
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --locked --release --target ${{ matrix.target }} ${{ matrix.features }}
+      - name: Running cargo build
+        run: cargo build --locked --target ${{ matrix.target }} ${{ matrix.features }}
```

Comment:
Is there a reason you went for `cargo build` instead of `cargo check`?

It seems that `cargo check` has some advantages in that it is a tad faster: https://doc.rust-lang.org/cargo/commands/cargo-check.html

Reply:
Cargo check is significantly faster than cargo build on the first run. After saving the cache, cargo build isn't that much worse. The documentation mentions that there are some errors cargo check can't detect. I thought for correctness cargo build could be better. Either one would work decently though.